src/app/earn/provider.tsx

```diff
@@ -106,6 +106,7 @@ export function EarnProvider(props: { children: any }) {
     updateQueryParams,
   } = useQueryParams({ ...defaultParams, network: chainIdRaw })
 
+  const [chainId, setChainId] = useState(chainIdRaw ?? MAINNET.chainId)
```

Comment:
We now have to keep track of `chainId` here because we need to be able to control it when setting tokens from/on different chains.